src/main/java/com/example/kidz_school/service/impl/UserServiceImpl.java

```diff
@@ -68,6 +70,84 @@ public void resetPassword(String email, String newPassword) {
         oneTimePasswordRepository.deleteByEmailEqualsIgnoreCase(email);
     }
 
+    @Override
+    public User save(User user) {
+        return userRepository.save(user);
+    }
+
+    public boolean isValid(UserRegistrationRequestDto userRequest, Model model) {
+        if (    !emailIsValid(userRequest.getEmail(), model) ||
+                !passwordIsValid(userRequest.getPassword(), userRequest.getRePassword(), model) ||
+                !nameIsValid(userRequest.getFirstName(), "First name", model) ||
+                !lastnameIsValid(userRequest.getLastName(), "Last name", model) ||
+                !phoneIsValid(userRequest.getPhone(), model)) {
+            return false;
+        }
```

Comment:
The request must be validated before it even gets to the endpoint. There are very useful annotations